### PR TITLE
MES-1367 - Dynamically Loading slots

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

Added the framework so we can load different slot components based on the vechicalSlotType

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/journal/journal.module.ts [hunk 1]
```diff
@@ -3,21 +3,38 @@ import { IonicPageModule } from 'ionic-angular';
...
+import { JournalIndicatorsComponent } from './journal-components/journal-indicators/journal-indicators';
+import { JournalTimeComponent } from './journal-components/journal-time/journal-time';
+import { JournalCandidateComponent } from './journal-components/journal-candidate/journal-candidate';
+import { JournalTestOutcomeComponent } from './journal-components/journal-test-outcome/journal-test-outcome';
 
 @NgModule({
   declarations: [
     JournalPage,
+    JournalTestDetailsComponent,
+		JournalIndicatorsComponent,
```
Comment (johnmorleybjss) [inline src/pages/journal/journal.module.ts]: Spacing is a little off here.

File: src/pages/journal/journal-components/journal-components.module.ts [hunk 2]
```diff
@@ -1,32 +0,0 @@
-import { NgModule } from '@angular/core';
```
Comment (johnmorleybjss) [inline src/pages/journal/journal-components/journal-components.module.ts]: I see you've pulled this module up into `journal.module.ts`. Is this due to some technical reason? Or is it just that having a component module per page not the preferred pattern?

File: src/providers/slot-selector/slot-item.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export class SlotItem {
+  constructor(public component: any, public slotData: any) {}
```
Comment (johnmorleybjss) [inline src/providers/slot-selector/slot-item.ts]: Could you do `public component: SlotComponent`?

File: src/pages/journal/journal-components/journal-components.module.ts (hunk 2 above)
Comment (Cavalier07) [reply src/pages/journal/journal-components/journal-components.module.ts]: Both :)

 In the how to lay out stuff meeting at the beginning we agreed to put all the components specific to a page in the page component.

Also when putting in the dynamic stuff in this PR we needed to put JournalTestSlot as an entry component into the Journal Page module, having the component in both modules started to bring down the walls between the 2 modules

File: src/pages/journal/journal.ts [hunk 4]
```diff
@@ -74,9 +81,19 @@ export class JournalPage extends BasePageComponent implements OnInit, OnDestroy
       isLoading$.pipe(map(this.handleLoadingSpinner))
     );
     this.createLoadingSpinner();
-    this.subscription = merged$.subscribe();
+    this.subscription = merged$.subscribe((data) => {
+      if (data) {
```
[the remark this answers is not in the document]
Comment (Cavalier07) [reply src/pages/journal/journal.ts]: Done

File: src/pages/journal/journal.ts [hunk 5]
```diff
@@ -74,7 +81,7 @@ export class JournalPage extends BasePageComponent implements OnInit, OnDestroy
       isLoading$.pipe(map(this.handleLoadingSpinner))
     );
     this.createLoadingSpinner();
-    this.subscription = merged$.subscribe();
+    this.subscription = merged$.subscribe((data) => this.createSlots(data));
```
Comment (ammarhaiderbjss) [inline src/pages/journal/journal.ts]: could it just be `.subscribe(this.createSlots)`?

Comment (Cavalier07) [reply src/pages/journal/journal.ts]: done :)

**Pull request merged**